Emby.Server.Implementations/Data/SqliteItemRepository.cs

```diff
@@ -3819,266 +3757,270 @@ private List<string> GetWhereClauses(InternalItemsQuery query, IStatement? state
                     }
 
-                    ratingClauseBuilder.Append("InheritedParentalRatingValue <= @MaxParentalRating");
+                    clauseBuilder.Append("InheritedParentalRatingValue <= @MaxParentalRating");
                     statement?.TryBind("@MaxParentalRating", query.MaxParentalRating.Value);
                 }
 
                 if (query.MinParentalRating.HasValue || query.MaxParentalRating.HasValue)
                 {
-                    ratingClauseBuilder.Append(")");
+                    clauseBuilder.Append(')');
                 }
 
                 if (!(query.MinParentalRating.HasValue || query.MaxParentalRating.HasValue))
                 {
-                    ratingClauseBuilder.Append(" OR InheritedParentalRatingValue not null");
+                    clauseBuilder.Append(" OR InheritedParentalRatingValue not null");
                 }
             }
             else if (query.MinParentalRating.HasValue)
             {
-                ratingClauseBuilder.Append("InheritedParentalRatingValue is null OR (InheritedParentalRatingValue >= @MinParentalRating");
+                clauseBuilder.Append("InheritedParentalRatingValue is null OR (InheritedParentalRatingValue >= @MinParentalRating");
                 statement?.TryBind("@MinParentalRating", query.MinParentalRating.Value);
 
                 if (query.MaxParentalRating.HasValue)
                 {
-                    ratingClauseBuilder.Append(" AND InheritedParentalRatingValue <= @MaxParentalRating");
+                    clauseBuilder.Append(" AND InheritedParentalRatingValue <= @MaxParentalRating");
                     statement?.TryBind("@MaxParentalRating", query.MaxParentalRating.Value);
                 }
 
-                ratingClauseBuilder.Append(")");
+                clauseBuilder.Append(')');
             }
             else if (query.MaxParentalRating.HasValue)
             {
-                ratingClauseBuilder.Append("InheritedParentalRatingValue is null OR InheritedParentalRatingValue <= @MaxParentalRating");
+                clauseBuilder.Append("InheritedParentalRatingValue is null OR InheritedParentalRatingValue <= @MaxParentalRating");
                 statement?.TryBind("@MaxParentalRating", query.MaxParentalRating.Value);
             }
             else if (!query.HasParentalRating ?? false)
             {
-                ratingClauseBuilder.Append("InheritedParentalRatingValue is null");
+                clauseBuilder.Append("InheritedParentalRatingValue is null");
             }
 
-            var ratingClauseString = ratingClauseBuilder.ToString();
-            if (!string.Equals(ratingClauseString, "(", StringComparison.OrdinalIgnoreCase))
+            if (clauseBuilder.Length > 1)
             {
-                whereClauses.Add(ratingClauseString + ")");
+                whereClauses.Add(clauseBuilder.Append(')').ToString());
+                clauseBuilder.Length = 0;
```

Comment:
Shouldn't we clear the clause builder even if length is 0?

Reply:
Shouldn't be necessary afaik

Reply:
All StringBuilder.Clear() does is setting length to 0

https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs,12bafac7fd9481fd

Reply:
I would argue the intent with  Clear is... Clearer

Reply:
I meant to say if the length is 1.. my mistake!

Reply:
@cvium I used .length = 0 instead of .clear bc it's done this way in the existing code, but I could change all occurrences if that's preferred.

Reply:
@Bond-009 Doesn't really matter to me